petab/lint.py

```diff
@@ -874,13 +916,34 @@ def assert_model_parameters_in_condition_or_parameter_table(
         parameter_df: PEtab parameter DataFrame
         model: PEtab model
         condition_df: PEtab condition table
+        mapping_df: PEtab mapping table
+        observable_df: PEtab observable table
+        measurement_df: PEtab measurement table
 
     Raises:
         AssertionError: in case of problems
     """
     allowed_in_condition_cols = set(model.get_valid_ids_for_condition_table())
+    if mapping_df is not None:
+        allowed_in_condition_cols |= {
+            from_id
+            for from_id, to_id in zip(mapping_df.index.values,
+                                      mapping_df[MODEL_ENTITY_ID])
+            # mapping table entities mapping to already allowed parameters
+            if to_id in allowed_in_condition_cols
+            # mapping table entities mapping to species
+            or model.is_state_variable(to_id)
```

Comment:
Okay, no, it makes sense the way it is. For SBML, `to_id in allowed_in_condition_cols` will always do the job. For PySB, the species ID itself is not allowed in the condition table, because they don't match our PEtab identifier requirements. That's why we have the mapping table. For this case, we need the `model.is_state_variable(to_id)
` part.